### PR TITLE
Update release process steps

### PR DESCRIPTION
**Pull request by aanm**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10035)

### CODE REVIEW

**Comment by aanm** [comment]
test-docs-please

File: Documentation/contributing/release/stable.rst [hunk 1]
```diff
@@ -92,6 +91,10 @@ If you intent to release a new feature release, see the
             [...]
             <<end of add-to-NEWS.rst>>
 
+#. Generate the release notes by running the instructions provided in github.com/cilium/release
```
Comment (joestringer) [inline Documentation/contributing/release/stable.rst:74]: This URL doesn't work, are you planning to push that repo before merging?

**Review by joestringer — APPROVED** [review]
Looks good. Won't backport cleanly to v1.5 because of docs refactoring. (Might not even backport to v1.6 for the same reason..)

File: Documentation/contributing/release/stable.rst (hunk 1 above)
Comment (aanm) [reply Documentation/contributing/release/stable.rst:74]: No, but the repo will have that name ;-)

**Review by joestringer — APPROVED** [review]

**Comment by aanm** [comment]
test-docs-please

**Comment by jrfastab** [comment]
@aanm I did some grep'ing in the 1.5 docs and can't find any of the phrases that would match these fixes. Any objection to dropping this from 1.5 Backports. If its important lets open an issue to refresh 1.5 documentation but I think its probably good-enough as-is and would focus more on current docs over contributing guides for 1.5

**Comment by jrfastab** [comment]
Actually I see what happened @raybejjani seems this is already merged we just missed updating tags?

**Comment by aanm** [comment]
@jrfastab yeah I think that's what happen

**Pull request merged**